assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js

```diff
@@ -0,0 +1,185 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useMemo, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+import { trackEvent } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import {
+  THEME_CONSTANTS,
+  Text,
+  Toggle,
+  Headline,
+} from '../../../../../../design-system';
+import { FullWidthWrapper } from '../../common/styles';
+import PillGroup from '../../shared/pillGroup';
+import localStore, {
+  LOCAL_STORAGE_PREFIX,
+} from '../../../../../utils/localStore';
+import useLibrary from '../../../useLibrary';
+import useStory from '../../../../../app/story/useStory';
+import {
+  getInUseFontsForPages,
+  getTextSetsForFonts,
+} from '../../../../../utils/getInUseFonts';
+import { Container as SectionContainer } from '../../../common/section';
+import { virtualPaneContainer } from '../../shared/virtualizedPanelGrid';
+import TextSets from './textSets';
+import { CATEGORIES, PANE_TEXT } from './constants';
+
+const TitleBar = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 20px 0;
+`;
+
+const TextSetsToggle = styled.div`
+  display: flex;
+  p {
+    margin: auto 12px;
+  }
+`;
+
+const TextSetsWrapper = styled.div`
+  ${virtualPaneContainer};
+`;
+
+function TextSetsPane({ paneRef }) {
+  const { textSets } = useLibrary(({ state: { textSets } }) => ({ textSets }));
+  const [showInUse, setShowInUse] = useState(false);
+
+  const allTextSets = useMemo(() => Object.values(textSets).flat(), [textSets]);
+  const storyPages = useStory(({ state: { pages } }) => pages);
+
+  const [selectedCat, setSelectedCat] = useState(
+    localStore.getItemByKey(`${LOCAL_STORAGE_PREFIX.TEXT_SET_SETTINGS}`)
+      ?.selectedCategory
+  );
+
+  const getTextSetsForInUseFonts = useCallback(
+    () =>
+      getTextSetsForFonts({
+        fonts: getInUseFontsForPages(storyPages),
+        textSets: selectedCat ? textSets[selectedCat] : allTextSets,
+      }),
+    [allTextSets, storyPages, selectedCat, textSets]
+  );
+
+  const addIdToTextSets = useCallback(
+    (sets) =>
+      sets.map((elements) => {
+        return {
+          id: `text_set_${uuidv4()}`,
+          elements,
+        };
+      }),
+    []
+  );
+  const filteredTextSets = useMemo(() => {
+    if (showInUse) {
+      return addIdToTextSets(getTextSetsForInUseFonts());
+    }
+    return addIdToTextSets(selectedCat ? textSets[selectedCat] : allTextSets);
+  }, [
+    selectedCat,
+    textSets,
+    allTextSets,
+    addIdToTextSets,
+    getTextSetsForInUseFonts,
+    showInUse,
+  ]);
+
+  const categories = useMemo(
+    () => [
+      ...Object.keys(textSets).map((cat) => ({
+        id: cat,
+        label: CATEGORIES[cat] ?? cat,
```

Comment:
💭 My preference is not to shorten variable names when we don't need to. I think it makes the code more readable 🐱 
```suggestion
      ...Object.keys(textSets).map((category) => ({
        id: category,
        label: CATEGORIES[category] ?? category,
```

Reply:
either works for me, i'll update it - shortening it doesn't do anything!